arch/arm/src/nrf52/nrf52_sdc.c

```diff
@@ -0,0 +1,609 @@
+  on_hci();
+  nxsem_post(&g_sdc_dev.exclsem);
+}
+
+/****************************************************************************
+ * Name: on_hci
+ ****************************************************************************/
+
+static void on_hci(void)
+{
+  struct bt_buf_s *outbuf;
+  size_t len;
+  int ret;
+  bool check_again = true;
+
+  while (check_again)
+    {
+      check_again = false;
+
+      /* Check for EVT by trying to get pending data into a generic
+       * buffer and then create an actual bt_buf_s, depending on msg length
+       */
+
+      ret = sdc_hci_evt_get(g_sdc_dev.msg_buffer);
+
+      if (ret == 0)
+        {
+          struct bt_hci_evt_hdr_s *hdr =
+              (struct bt_hci_evt_hdr_s *)g_sdc_dev.msg_buffer;
+
+          len = sizeof(*hdr) + hdr->len;
+
+#ifdef CONFIG_DEBUG_WIRELESS_INFO
+          if (hdr->evt == BT_HCI_EVT_CMD_COMPLETE)
+            {
+              struct hci_evt_cmd_complete_s *cmd_complete =
+                  (struct hci_evt_cmd_complete_s *)
+                      (g_sdc_dev.msg_buffer + sizeof(*hdr));
+              uint8_t *status = (uint8_t *)cmd_complete + 1;
+
+              wlinfo("received CMD_COMPLETE from softdevice "
+                     "(opcode: %d, status: 0x%x)\n",
+                     cmd_complete->opcode, *status);
+            }
+          else
+            {
+              wlinfo("received HCI EVT from softdevice "
+                     "(evt: %d, len: %zu)\n", hdr->evt, len);
+            }
+#endif
+
+          outbuf = bt_buf_alloc(BT_EVT, NULL, BLUETOOTH_H4_HDRLEN);
+          bt_buf_extend(outbuf, len);
+
+          memcpy(outbuf->data, g_sdc_dev.msg_buffer, len);
+
+          bt_hci_receive(outbuf);
+
+          check_again = true;
+        }
+
+      /* Same for ACL */
+
+      ret = sdc_hci_data_get(g_sdc_dev.msg_buffer);
+
+      if (ret == 0)
+        {
+          struct bt_hci_acl_hdr_s *hdr =
+              (struct bt_hci_acl_hdr_s *)g_sdc_dev.msg_buffer;
+
+          wlinfo("received HCI ACL from softdevice (handle: %d)\n",
+                 hdr->handle);
+
+          len = sizeof(*hdr) + hdr->len;
+
+          outbuf = bt_buf_alloc(BT_ACL_IN, NULL, BLUETOOTH_H4_HDRLEN);
+          bt_buf_extend(outbuf, len);
+
+          memcpy(outbuf->data, g_sdc_dev.msg_buffer, len);
+
+          bt_hci_receive(outbuf);
+
+          check_again = true;
+        }
+    }
+}
+
+/****************************************************************************
+ * Name: swi_isr
+ ****************************************************************************/
+
+static int swi_isr(int irq, FAR void *context, FAR void *arg)
+{
+  work_queue(LPWORK, &g_sdc_dev.work, low_prio_worker, NULL, 0);
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: rng_handler
+ ****************************************************************************/
+
+static void rng_handler(void)
+{
+  sdc_RNG_IRQHandler();
```

Comment:
I don't have a strong feeling here (ignoring is fine), but one thing we could do to help with the mix case error where we are calling a library function is to map them using a define in a header.  Then we only have a single file that has the violations we have to deal with.
```
#define sdc_rng_irqhandler()  sdc_RNG_IRQHandler()
```

Reply:
Wasn't there a white list in nxstyle somewhere? If not, I can follow your suggestion.

Reply:
I think the whitelist was opposed because it opens a slippery slope of style violations.  The workaround I proposed is just to make it so we don't keep having to triage every file that uses the interface and people are less likely to pick up a bad habit.  This is purely a suggestion, I am fine as is.

Reply:
I understand. For now I will leave it as is, until MPSL/SDC is used somewhere else, which I'm not sure is very likely.
When that happens we can do as you suggest.